yt/frontends/boxlib/data_structures.py

```diff
@@ -1650,7 +1650,7 @@ def _parse_parameter_file(self):
                     self.parameters[l[0].strip()] = l[1].strip()
 
         # set the periodicity based on the integer BC runtime parameters
-        self.periodicity[:] = False
+        self.periodicity = (False, False, False)
         if "geometry.is_periodic" in self.parameters:
             is_periodic = self.parameters["geometry.is_periodic"].split()
             self.periodicity[: len(is_periodic)] = [val == "1" for val in is_periodic]
```

Comment:
This won't work either because tuples are immutable, here's a fix (backpedaling again...)
```suggestion
        self.periodicity = [False, False, False]
        if "geometry.is_periodic" in self.parameters:
            is_periodic = self.parameters["geometry.is_periodic"].split()
            self.periodicity[: len(is_periodic)] = [val == "1" for val in is_periodic]
        self.periodicity = ensure_tuple(self.periodicity)
```

Reply:
applied as code comment.